### ISSUE TITLE
pnext: Support wait for log write on commit in code host.

### ISSUE THREAD

**Issue by eriohl** [Starcounter/Starcounter#3582]
Commit will still be synchronous but context will be released for use by another task while waiting for pending log write.

Requires integration between DM asynchronous callback and legacy task scheduler.

_Dependent on adaption to metadata layer_

**Comment by eriohl** [comment]
Possible issue is that per scheduler thread count is limited. Not sure how running into this limit is handled in the code host currently.

**Comment by eriohl** [comment]
Another complication is that metadata operations creates their own transaction. Combining this with asynchronous commit is problematic since the wait and signalling is host specific while metalayer is not. Generally however these operations create only one transaction that is committed at the end of the operation which should make it possible to treat the entire operation the same way as an asynchronous commit. There are exceptions however where multiple transactions are created and committed in a single operation.

**Comment by eriohl** [comment]
The issue with transactions in metadata layer before we can proceed on this. Either it needs to be restructured so that there is only one transaction per operation so that the operation is self can be treated as a asynchronous operation or it will require some interaction with the host (personally I favor the former).

Creation of metadata tables is the biggest concern since this operation requires multiple commits to function. On the other hand, this should not be done on the host but be done on database initialization before any IWP can be bound. So in the long run this can be solved by moving this code to its proper place.

**Comment by eriohl** [comment]
Update: Only metadata table creation actually need multiple commits.

**Comment by bigwad** [comment]
done
